src/profiler.h

```diff
@@ -58,54 +58,71 @@ class Profiler
 	void getPage(GraphValues &o, u32 page, u32 pagecount);
 
 
+	void graphSet(const std::string &id, float value)
+	{
+		MutexAutoLock lock(m_mutex);
+		m_graphvalues[id] = value;
+	}
 	void graphAdd(const std::string &id, float value)
 	{
 		MutexAutoLock lock(m_mutex);
-		std::map<std::string, float>::iterator i =
-				m_graphvalues.find(id);
-		if(i == m_graphvalues.end())
-			m_graphvalues[id] = value;
+		auto it = m_graphvalues.find(id);
+		if (it == m_graphvalues.end())
+			m_graphvalues.emplace(id, value);
 		else
-			i->second += value;
+			it->second += value;
 	}
 	void graphGet(GraphValues &result)
 	{
 		MutexAutoLock lock(m_mutex);
-		result = m_graphvalues;
-		m_graphvalues.clear();
+		assert(result.empty());
+		std::swap(result, m_graphvalues);
 	}
 
 	void remove(const std::string& name)
 	{
 		MutexAutoLock lock(m_mutex);
-		m_avgcounts.erase(name);
 		m_data.erase(name);
 	}
 
 private:
+	struct DataPair {
```

Comment:
A constructor might make sense here so that we can be sure that the members are always initialized.